### ISSUE TITLE
n:macros not available in <script type="text/html">

### ISSUE THREAD

**Issue by xpavp03** [nette/latte#71]
```latte
<script type="text/html">
<a n:href="Page: param">fails with n:macro</a>
<a href="{plink Page: param}">works with plink</a>
</script>
```
will be rendered as
```html
<script type="text/html">
<a n:href="Page: param">fails with n:macro</a>
<a href="/param">works with plink</a>
</script>
```
I didn't test all n:macros, only n:href and n:class.

**Comment by xpavp03** [comment]
It's possible to work around the issue by adding `{contentType html}` inside of `<script>`.

```latte
<script type="text/html">
{contentType html}
```

**Comment by fprochazka** [comment]
What is your point? Are you bugreporting? Or are you requesting feature?

**Comment by JanTvrdik** [comment]
Well I personally thought that `type="text/html"` [would do](https://github.com/nette/latte/blob/100ecb48ccbd44cde5b7b550daafac2c771f6957/src/Latte/Compiler.php#L328-L329) the same as `{contentType html}`, maybe it is a bug.

**Comment by xpavp03** [comment]
@fprochazka Yes, I consider this to be a bug. Why would you think otherwise?

**Comment by JanTvrdik** [comment]
@xpavp03 Because current processing is identical to what browser does.

**Comment by xpavp03** [comment]
@JanTvrdik I don't understand your answer. Browser doesn't render Latte at all so even {$var} wouldn't work. {$var} currently does work in `<script type="text/html">`.

And why should it be the goal anyway? I guess in some cases we might need to turn off Latte processing completely and for that I assume syntax="off" exists.

I don't see the benefit of only disabling n:macros in `<script type="text/html">`.

**Comment by JanTvrdik** [comment]
> I don't understand your answer. 

You believe that inside `<script type="text/html">` are some HTML tags with n:attributes that do not work. But the problem is that there are no HTML tags at all. Both browser and Latte parser treats the content of `<script>` and `<style>` as CDATA.

This is the reason why the following code in valid:

~~~html
<script>
var foo = "<unknownTag>";
</script>
~~~

Neither browser nor Latte recognize `<unknownTag>` as HTML tag. Changing this behavior is not so simple. Consider the following code:

~~~html
<script type="text/html">
    INSIDE SCRIPT TAG
    <script type="application/javascript">

    </script>
     THIS IS NOT IN SCRIPT TAG
</script>
~~~

> I don't see the benefit of only disabling n:macros 

They are not disabled but there are not HTML tags they can be attach to.

**Comment by xpavp03** [comment]
Oh, I understand now. Thanks. Mhm...
